ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/ITExplorerTest.java

```diff
@@ -0,0 +1,222 @@
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.nuxeo.functionaltests.AbstractTest;
+import org.nuxeo.functionaltests.Locator;
+import org.nuxeo.functionaltests.RestHelper;
+import org.nuxeo.functionaltests.explorer.pages.ArtifactHomePage;
+import org.nuxeo.functionaltests.explorer.pages.ArtifactPage;
+import org.nuxeo.functionaltests.explorer.pages.ExplorerHomePage;
+import org.nuxeo.functionaltests.pages.DocumentBasePage.UserNotConnectedException;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+/**
+ * Test explorer main webengine pages.
+ *
+ * @since 11.1
+ */
+public class ITExplorerTest extends AbstractTest {
+
+    @Before
+    public void before() {
+        RestHelper.createUser(TEST_USERNAME, TEST_PASSWORD, null, null, null, null, "members");
+    }
+
+    @After
+    public void after() {
+        RestHelper.cleanup();
+    }
+
+    protected void doLogin() {
+        getLoginPage().login(TEST_USERNAME, TEST_PASSWORD);
+    }
+
+    protected void doLogout() {
+        // logout avoiding JS error check
+        driver.get(NUXEO_URL + "/logout");
+    }
+
+    protected ExplorerHomePage goHome() {
+        open(ExplorerHomePage.URL);
+        return asPage(ExplorerHomePage.class);
+    }
+
+    /**
+     * Simple login, logout test, checking the home page is displayed without errors after login.
+     */
+    @Test
+    public void testLoginLogout() throws UserNotConnectedException {
+        doLogin();
+        goHome();
+        doLogout();
+    }
+
+    @Test
+    public void testHomePage() throws UserNotConnectedException {
+        doLogin();
+        ExplorerHomePage home = goHome();
+        assertEquals("Nuxeo Platform Explorer", home.getTitle());
+        assertEquals("Running Platform".toUpperCase(), home.currentPlatform.getText());
+
+        ArtifactHomePage ahome = home.navigateTo(home.currentExtensionPoints);
+        assertEquals("All extension points", ahome.getTitle());
+        ahome = goHome().navigateTo(home.currentContributions);
+        assertEquals("All contributions", ahome.getTitle());
+        ahome = goHome().navigateTo(home.currentExtensionPoints);
+        assertEquals("All extension points", ahome.getTitle());
+        ahome = goHome().navigateTo(home.currentOperations);
+        assertEquals("All operations", ahome.getTitle());
+        ahome = goHome().navigateTo(home.currentServices);
+        assertEquals("All services", ahome.getTitle());
+
+        doLogout();
+    }
+
+    @Test
+    public void testExtensionPoints() throws UserNotConnectedException {
+        doLogin();
+        ExplorerHomePage home = goHome();
+        ArtifactHomePage ahome = home.navigateTo(home.currentExtensionPoints);
+        assertTrue(ahome.isSelected(ahome.extensionPoints));
+        WebElement elt = ahome.getFirstListingElement();
+        WebElement link = elt.findElement(By.xpath(".//a"));
```

Comment:
What comes first in my mind seeing all those tests; you should use some Class (like `ArtifactHomePage`) and accessors to make it more understandable instead of relying on direct selectors to `By.xpath` and consosr.

Reply:
I agree it is not that consistent: i just wanted to have quick ftests here, because the selectors depend on the page, so i'd rather make ftl files more consistent to ease up test classes maintenance (that's why i kept it kind of hardcoded here).

Also note i'll make some visual changes with https://jira.nuxeo.com/browse/NXP-28913, so i can try to work on this at the same time if it makes sense.

Reply:
will do with https://jira.nuxeo.com/browse/NXP-28913

Reply:
see https://github.com/nuxeo/nuxeo/pull/3959